89.md

```diff
@@ -0,0 +1,112 @@
+NIP-89
+======
+
+Recommended Application Handlers
+--------------------------------
+
+`draft` `optional` `author:pablof7z`
+
+This NIP describes `kind:31989` and `kind:31990`: a way to discover applications that can handle unknown event-kinds.
+
+## Rationale
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for clients to discover applications that handle events of a specific kind to ensure smooth cross-client and cross-kind interactions.
+
+### Parties involved
+There are three actors to this workflow:
+
+* application that handles a specific event kind (note that an application doesn't necessarily need to be a distinct entity and it could just be the same pubkey as user A)
+    * Publishes `kind:31990`, detailing how apps should redirect to it
+* user A, who recommends an app that handles a specific event kind
+    * Publishes `kind:31989`
+* user B, who seeks a recommendation for an app that handles a specific event kind
+    * Queries for `kind:31989` and, based on results, queries for `kind:31990`
+
+# Events
+
+## Recommendation event
+```json
+{
+    "kind": 31989,
+    "pubkey": <recommender-user-pubkey>,
+    "tags": [
+        [ "d", <supported-event-kind> ],
+        [ "a", "31990:app1-pubkey:<d-identifier>", "wss://relay1", "ios" ],
+        [ "a", "31990:app2-pubkey:<d-identifier>", "wss://relay2", "web" ]
+    ]
+}
+```
+
+Multiple `a` tags can appear on on the same `kind:31989`.
+
+The second value of the tag SHOULD be a relay hint.
+The third value of the tag SHOULD be the platform where this recommendation might apply.
+
+## Handler information
+```json
+{
+    "kind": 31990,
+    "pubkey": <application-pubkey>,
+    "content": "<optional-kind:0-style-metadata>",
+    "tags": [
+        [ "d", <random-id> ],
+        [ "k", <supported-event-kind> ],
+        [ "web", "https://..../a/<bech32>", "nevent" ],
+        [ "web", "https://..../p/<bech32>", "nprofile" ],
+        [ "web", "https://..../e/<bech32>" ],
+        [ "ios", ".../<bech32>" ]
+    ]
+}
+```
+
+* `content` is an optional `set_metadata`-like stringified JSON object, as described in NIP-01. This content is useful when the pubkey creating the `kind:31990` is not an application. If `content` is empty, the `kind:0` of the pubkey should be used to display application information (e.g. name, picture, web, LUD16, etc.)
+
+* `k` tags' value is the event kind that is supported by this `kind:31990`.
+Multiple `k` tags can exist in the same event if the application supports more than one event kind.
+
+* `bech32` in a URL MUST be replaced by clients with the NIP-19-encoded entity that should be loaded by the application.
+
+Multiple tags might be registered by the app, following NIP-19 nomenclature as the second value of the array.
+
+A tag without a second value in the array SHOULD be considered a generic handler for any NIP-19 entity that is not handled by a different tag.
+
```

Comment:
Maybe add that this model allows for several 31990 infos per app, which could be used to provide different handler urls, or different app infos.